doc/c-style-guide.md

```diff
@@ -0,0 +1,41 @@
+GRPC C STYLE GUIDE
+=====================
+
+Background
+----------
+
+Here we document style rules for C usage in the gRPC Core library.
+
+General
+-------
+
+- Layout rules are defined by clang-format, and all code should be passed through
+  clang-format. A (docker-based) script to do so is included in 
+  tools/distrib/clang_format_code.sh.
+
+Header Files
+------------
+
+- Public header files (those in the include/grpc tree) should compile as pedantic C89
```

Comment:
I'd like to keep the include/ tree C89.

My rationale:
suppose you have a C code base that you'd like not to transition to C99 for whatever reason, but you want to use gRPC. It's likely you can compile a .a with gRPC using a C99 compiler, and rely on just the C89 headers from your own codebase.